### PR TITLE
Issue #295 Ensure Jetty Client use of Deflater calls .end() to avoid memory leak

### PR DESCRIPTION
**Pull request by prodansergiu**

### CODE REVIEW

File: jetty-client/src/main/java/org/eclipse/jetty/client/GZIPContentDecoder.java [hunk 1]
```diff
@@ -322,6 +323,11 @@ private void reset()
         flags = 0;
     }
 
+    @Override
+    public void destroy() {
```
Comment (sbordet) [inline jetty-client/src/main/java/org/eclipse/jetty/client/GZIPContentDecoder.java]: Please follow formatting of surrounding code: braces on new lines.

File: jetty-client/src/main/java/org/eclipse/jetty/client/HttpReceiver.java [hunk 2]
```diff
@@ -465,6 +466,9 @@ private void terminateResponse(HttpExchange exchange, Result result)
      */
     protected void reset()
     {
+        if (decoder instanceof Destroyable) {
+            ((Destroyable) decoder).destroy();
```
Comment (sbordet) [inline jetty-client/src/main/java/org/eclipse/jetty/client/HttpReceiver.java]: No space after cast.

File: jetty-client/src/main/java/org/eclipse/jetty/client/HttpReceiver.java [hunk 3]
```diff
@@ -477,6 +481,9 @@ protected void reset()
      */
     protected void dispose()
     {
+        if (decoder instanceof Destroyable) {
+            ((Destroyable) decoder).destroy();
+        }
```
Comment (sbordet) [inline jetty-client/src/main/java/org/eclipse/jetty/client/HttpReceiver.java]: Please refactor the destruction of the decoder into a method, say `destroyDecoder(ContentDecoder)`, and call this new method from both `reset()` and `dispose()`

**Comment by prodansergiu** [comment]
Should also change the commit message to "Ensure Jetty Client use of Inflater calls .end() to avoid memory leak" instead of the original version "Deflater"

**Comment by sbordet** [comment]
Thanks !

**Comment by prodansergiu** [comment]
Thanks for the review and quick response! :+1:

**Pull request merged**